web/react/components/post_list.jsx

```diff
@@ -128,6 +128,21 @@ export default class PostList extends React.Component {
                 this.userHasSeenNew = true;
             }
             this.isUserScroll = true;
+
+            $('.selected-post').removeClass('selected-post');
+
+            $(React.findDOMNode(this.refs.postlistcontent)).children().each(function select() {
+                if ($(this).position().top > 0) {
+                    if (!$('.selected-post')[0]) {
+                        $(this).addClass('selected-post');
+                    }
+                    return;
+                }
+
+                if ($(this).position().top + $(this).height() / 2 > 0) {
+                    $(this).addClass('selected-post');
+                }
+            });
```

Comment:
1) Only one post will be marked 'selected-post' at a time and yes it will be the top-most visible post. 
2) I'm 0/5 on the naming
3) The return just skips the second check since it will be pointless if the first one is true.

The code goes through each post DOM node and checks first if the top is visible. If it is then we only want to mark it as the selected post if no other post has been selected yet. The second check is to see if at least half the post is visible, and select it if it is.